### PR TITLE
Conditional link to job manager

### LINKED ISSUE #1258: Handle workspaces that have not been migrated to Job Manager
Whether a workspace's jobs show up in job manager is a workspace-by-workspace condition. Workspaces that show their jobs in Job Manager carry a special "workflow-collection" label in their workspace metadata.

If this label is present in the workspace, submission links in the job history should go to job manager.

If it is not, they should continue to go to FireCloud. Beforehand we should present a confirmation screen messaging what is happening:
"We are currently introducing Terra's new job management component for accessing workflow details. However, this workspace isn't yet ready to use the new job manager. For now, workflow details can be found in our legacy system, FireCloud. You will be asked to sign in to FireCloud to view your workflow details.
Button: Go To FireCloud"

We would need to implement this change in time for the job manager rollout, for which the earliest date is 3/12.

Old workspaces will slowly have the workflow-collection label added to them as part of a multiweek migration. When this migration is complete, we can rip this logic out.

### PR DESCRIPTION
**Pull request by zarsky-broad**
(once it's live)

Closes #1258

### CODE REVIEW

**Review by breilly2 — APPROVED** [review]
👍 Nicely done!

File: src/pages/workspaces/workspace/JobHistory.js [hunk 1]
```diff
@@ -267,6 +271,20 @@ const JobHistory = _.flow(
           Utils.formatNumber(collapsedStatuses(_.find({ submissionId: aborting }, submissions).workflowStatuses).running)
         } running workflow(s)?`
       ]),
+      linkToFC && h(Modal, {
+        onDismiss: () => this.setState({ linkToFC: undefined }),
+        title: 'Legacy Workflow Details',
```
Comment (katevoss) [inline src/pages/workspaces/workspace/JobHistory.js:276]:
```suggestion
        title: 'Legacy Job Details',
```

File: src/pages/workspaces/workspace/JobHistory.js [hunk 2]
```diff
@@ -267,6 +271,20 @@ const JobHistory = _.flow(
...
+        okButton: buttonPrimary({
+          as: 'a',
+          href: linkToFC,
+          target: '_blank',
+          onClick: () => this.setState({ linkToFC: undefined })
+        }, 'Go To FireCloud')
+      }, [
+        `We are currently introducing Terra's new job management component for accessing workflow details. However, this
```
Comment (katevoss) [inline src/pages/workspaces/workspace/JobHistory.js:284]:
```suggestion
        `We are introducing Terra's new job management page for accessing workflow details. However, this 
```

File: src/pages/workspaces/workspace/JobHistory.js [hunk 3]
```diff
@@ -267,6 +271,20 @@ const JobHistory = _.flow(
...
+        title: 'Legacy Workflow Details',
+        okButton: buttonPrimary({
+          as: 'a',
+          href: linkToFC,
+          target: '_blank',
+          onClick: () => this.setState({ linkToFC: undefined })
+        }, 'Go To FireCloud')
+      }, [
+        `We are currently introducing Terra's new job management component for accessing workflow details. However, this 
+        workspace isn't yet ready to use the new job manager. For now, workflow details can be found in our legacy system,
```
Comment (katevoss) [inline src/pages/workspaces/workspace/JobHistory.js:285]:
```suggestion
        workspace isn't ready yet to use the job manager. In the meantime, you can find workflow details in our legacy system, 
```

File: src/pages/workspaces/workspace/JobHistory.js [hunk 4]
```diff
@@ -267,6 +271,20 @@ const JobHistory = _.flow(
...
+        okButton: buttonPrimary({
+          as: 'a',
+          href: linkToFC,
+          target: '_blank',
+          onClick: () => this.setState({ linkToFC: undefined })
+        }, 'Go To FireCloud')
+      }, [
+        `We are currently introducing Terra's new job management component for accessing workflow details. However, this 
+        workspace isn't yet ready to use the new job manager. For now, workflow details can be found in our legacy system, 
+        FireCloud. You will be asked to sign in to FireCloud to view your workflow details.`
```
Comment (katevoss) [inline src/pages/workspaces/workspace/JobHistory.js:286]:
```suggestion
        FireCloud, by signing in with the same username and password that you use in Terra.`
```

**Review by katevoss — COMMENTED** [review]
A few suggestions on the wording. How can I test this out?

**Comment by zarsky-broad** [comment]
@katevoss to test, run a tool in a new workspace and run this locally, switching the boolean from the top of the file.

@bradtaylor thoughts on wording?

File: src/pages/workspaces/workspace/JobHistory.js [hunk 5]
```diff
@@ -165,17 +166,20 @@ const JobHistory = _.flow(
...
-                      link({
+                      link(viewInJobManager ? {
                         target: '_blank',
-                        //check link with actual job manager link
-                        href: linkToJobManager ? `${getConfig().jobManagerUrlRoot}?q=submission-id%3D${submissionId}` :
-                          `${getConfig().firecloudUrlRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
+                        href: `${getConfig().jobManagerUrlRoot}?q=submission-id%3D${submissionId}`
+                      } : {
+                        onClick: () => this.setState({
+                          linkToFC: `${getConfig().firecloudUrlRoot}/#workspaces/${namespace}/${name}/monitor/${submissionId}`
```
Comment (panentheos) [inline src/pages/workspaces/workspace/JobHistory.js:181]: nit: We could just store the `submissionId` in state, and compute the full URL dynamically.

**Review by panentheos — APPROVED** [review]
Looks good.

**Comment by zarsky-broad** [comment]
@panentheos true, but this is all temporary code, so I'm not gonna bother

**Pull request merged**